src/ctx.rs

```diff
@@ -75,6 +76,37 @@ pub fn small_memset<T: Clone + Copy, const UP_TO: usize, const WITH_DEFAULT: boo
     }
 }
 
+/// # Safety
+///
+/// `buf` must be correctly aligned and dereferencable (but need not be
+/// initialized). `T` must not have a destructor.
+fn small_memset_raw<T: Clone + Copy, const UP_TO: usize, const WITH_DEFAULT: bool>(
+    buf: *mut [T],
+    val: T,
+    offset: usize,
+    len: usize,
+) {
+    //assert!(buf.len() >= offset && buf.len() - offset >= len);
```

Comment:
Why's this commented out?

Reply:
Good catch, was checking performance. It's ok with it in place.